### PR TITLE
StatusHistory: Support legend decimals based on field overrides

### PR DESCRIPTION
**Pull request by nmarrs**
Related to https://github.com/grafana/grafana/issues/72485

The status history legend is currently hardcoded to only display integer values. This PR allows the user to set the desired decimals through field overrides. Note: all y axis fields must have this override applied for this to be supported.

Ideally the standard option decimal setting would apply globally to both the displayed values in the visualization as well as the legend. See the related issue above for on this.

Before
image

After
Field override fixes  decimals

[Debug_ Example Status History __ 2023-07-28 02_05_15-1690518716821.json.txt](https://github.com/grafana/grafana/files/12190235/Debug_.Example.Status.History.__.2023-07-28.02_05_15-1690518716821.json.txt)

### CODE REVIEW

File: public/app/core/components/TimelineChart/utils.ts [hunk 1]
```diff
@@ -525,9 +525,9 @@ export function getThresholdItems(fieldConfig: FieldConfig, theme: GrafanaTheme2
   }
 
   const steps = thresholds.steps;
-  const disp = getValueFormat(thresholds.mode === ThresholdsMode.Percentage ? 'percent' : fieldConfig.unit ?? '');
+  const getDisplay = getValueFormat(thresholds.mode === ThresholdsMode.Percentage ? 'percent' : fieldConfig.unit ?? '');
 
-  const fmt = (v: number) => formattedValueToString(disp(v));
+  const format = (value: number) => formattedValueToString(getDisplay(value, fieldConfig.decimals ?? 0));
```
Comment (ryantxu) [inline public/app/core/components/TimelineChart/utils.ts:530]: do you want to default to 0?  i think undefined will pick the auto decimals when not set

Comment (nmarrs) [reply public/app/core/components/TimelineChart/utils.ts:530]: I'll test this out locally and see what the behavior is 👍

Comment (leeoniya) [reply public/app/core/components/TimelineChart/utils.ts]: this second arg wasnt really meant for this, but should work :)

have not looked at the code, but why wouldnt the regular displayProcessor respect the decimals specified in the options/fieldconfig without needing this additional override? 🤔

**Review by drew08t — APPROVED** [review]
LGTM and tested locally!

**Pull request merged**